### ISSUE TITLE
Duplicate IDs in sprites

### ISSUE THREAD

**Issue by whatcould** [karify/external-svg-sprite-loader#27]
I ran into a frustrating issue that might be particular to my SVG files (exported by Sketch); the bug was caused by putting them all together as a single sprite file. I looked at SVGO options but wasn't able to solve it. Would like to figure out if there's a better/permanent fix so others don't have this issue...

**The symptom** was that my icons were coming out all duplicated — the first one loaded seemed to be repeating for every image instance.

**The cause** was that the `<path>` elements inside each `<symbol>` had repeating IDs — see below how `id="a"` repeats (with the corresponding `<use>` reference to `#a`). The original Sketch exports, before the SVGO parsing, had IDs of `id="path-1"` instead of all `id="a"`, but that still would have broken when combined into one XML file. 

```xml
<symbol id="icon-play-a6b2d" viewBox="0 0 24 24">
  <defs>
    <path d="..." id="a"/>
  </defs>
  <use xlink:href="#a" transform="translate(2)" fill-rule="evenodd"/>
</symbol>
<symbol id="icon-pause-c28a5" viewBox="0 0 24 24">
  <defs>
    <path d="..." id="a"/>
  </defs>
  <use xlink:href="#a" transform="translate(4 1)" fill-rule="evenodd"/>
</symbol>
... repetition continues
```

I was able to fix my particular issue by going through the individual symbols in `createDefs` within `SvgDocument.js` and doing a simple `.replace` to create unique IDs based on the index (see below). This worked with my SVG, but I'd be nervous it might mess up more complicated SVG documents?

```js
    static createDefs(...contents) {
      contents = contents.map( (symbol, index) => {
        const id = `a-${index}`
        return symbol.replace('xlink:href="#a"', `xlink:href="#${id}"`).replace('id="a"/>', `id="${id}"/>`)
      })
        return `<defs>${contents.join('')}</defs>`;
    }
```

**Comment by bensampaio** [comment]
@whatcould this is indeed a problem other people are also experiencing. There is a solution for it in #29 which should fix all cases. Hopefully it will get merged soon :)

**Comment by stowball** [comment]
[My PR](https://github.com/karify/external-svg-sprite-loader/pull/29) won't fix this scenario as it didn't touch the `<use xlink:href />`. TBH, I'm not sure why the `<use>` is even output in the plugin; that should be the responsibility of the implementor in the markup, no?

**Comment by bensampaio** [comment]
The `<use>` is necessary so that you can reference the sprite in CSS

**Comment by bensampaio** [comment]
@whatcould could you share one of the SVGs you are having problems with? I am trying to fix this but I need to be able to test it.

**Comment by bensampaio** [comment]
I believe this should be fixed in 2.3.0. Let me know if that's not the case.

**Comment by whatcould** [comment]
Looks like it works great! Thanks for addressing this.
